bin/process-node-options.js

```diff
@@ -0,0 +1,44 @@
+
+const spawn = require("cross-spawn");
+const path = require("path");
+const args = [path.join(__dirname, "webpack")];
+
+/**
+ * Process arguments and spawns a node process applying node/v8 options
+ * @param {array} argvs webpack-cli arguments
+ * @returns {void}
+ */
+module.exports = (argvs) => {
+	argvs.slice(2).forEach(arg => {
+		const normalizedArg = arg.replace("node.", "");
+		if (arg.includes("node.")) {
+			args.unshift(normalizedArg);
+		} else {
+			args.push(arg);
+		}
+	});
+
+	const webpackCliProcess = spawn(process.execPath, args, {
+		stdio: "inherit"
+	});
+
+	webpackCliProcess.on("exit", (code, signal) => {
+		if (signal) {
+			process.kill(process.pid, signal);
+		} else {
+			process.exit(code);
+		}
+	});
+
+	// when process was terminated ([Ctrl/Cmd]+C)
+	process.on("SIGINT", () => {
+		webpackCliProcess.kill("SIGINT");
+		webpackCliProcess.kill("SIGTERM");
+	});
+
+	// when main process was exited.
+	process.on("exit", (code) => {
```

Comment:
This was the part "missing" @ev1stensberg , the other one above just relied on `Ctrl/Cmd+C` which sometimes was not the case. I've compared twice master branch and this branch, both are running almost identical (spawning 1 process, when tests for node-flags are running) in terms of process being spawned and this branch is not spawning billions anymore.